pkg/state/state_shared_dp.go

```diff
@@ -65,10 +65,10 @@ func (s *stateSharedDp) Sync(customResource interface{}, infoCatalog InfoCatalog
 	log.V(consts.LogLevelInfo).Info(
 		"Sync Custom resource", "State:", s.name, "Name:", cr.Name, "Namespace:", cr.Namespace)
 
-	if cr.Spec.DevicePlugin == nil {
+	if cr.Spec.RdmaSharedDevicePlugin == nil {
 		// Either this state was not required to run or an update occurred and we need to remove
 		// the resources that where created.
-		// TODO: Support the latter case
+		// TODO: Support default configuration should be applied
```

Comment:
what are the default configurations ?

the original intention was that if this is nil then this state should be "undone" i.e removal of all yaml in this state's manifest folder